cpp/inc/bond/core/detail/extern_macro.h

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <bond/core/config.h>
+
+#if !defined(BOND_COMPACT_BINARY_PROTOCOL) \
+ && !defined(BOND_SIMPLE_BINARY_PROTOCOL) \
+ && !defined(BOND_FAST_BINARY_PROTOCOL) \
+ && !defined(BOND_SIMPLE_JSON_PROTOCOL)
+#error None of the built-in protocols is enabled
```

Comment:
What about a message like this?

"None of the built-in protocols are enabled. Either enable one of the built-in protocols or set BOND_LIB_TYPE to BOND_LIB_TYPE_HEADER."